onnxruntime/test/providers/vitisai/vitisai_basic_test.cc

```diff
@@ -0,0 +1,316 @@
+#pragma comment(lib, "d3d12.lib")
+
+#include "core/providers/vitisai/imp/tensor_proto.h"
+
+#include <wil/result.h>
+#include <gtest/gtest.h>
+#include <cstddef>
+
+#include <iostream>
+#include <string>
+#include <fstream>
+
+using namespace vaip;
+using namespace std;
+using Microsoft::WRL::ComPtr;
+
+namespace onnxruntime {
+namespace test {
+
+LARGE_INTEGER getStartingTime() {
+  LARGE_INTEGER startingTime;
+  QueryPerformanceCounter(&startingTime);
+  return startingTime;
+}
+
+int getElapsedTime(LARGE_INTEGER startingTime) {
+  LARGE_INTEGER endingTime, elapsedMicroseconds, frequency;
+  QueryPerformanceFrequency(&frequency);
+  QueryPerformanceCounter(&endingTime);
+  elapsedMicroseconds.QuadPart = endingTime.QuadPart - startingTime.QuadPart;
+  //
+  // We now have the elapsed number of ticks, along with the
+  // number of ticks-per-second. We use these values
+  // to convert to the number of elapsed microseconds.
+  // To guard against loss-of-precision, we convert
+  // to microseconds *before* dividing by ticks-per-second.
+  //
+  elapsedMicroseconds.QuadPart *= 1000000;
+  elapsedMicroseconds.QuadPart /= frequency.QuadPart;
+
+  return elapsedMicroseconds.QuadPart;
+}
+
+struct uploadReadback {
+  float upload, readback;
+};
+
+typedef struct uploadReadback Struct;
+
```

Comment:
That is an unnecessary and hard-to-read name - it hides the purpose. Use the original name `uploadReadback`. In C++, you don't need to prepend it with the `struct` keyword. So use `uploadReadback and not `struct uploadReadback`.